arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx

```diff
@@ -266,7 +267,123 @@ Note the `logs` field within the transaction receipt:
       ]
 ```
 
-From the `logs` field, we can see the `topics[0]` is `0xa937..cfa`, which is the event signature of `Hi(address)`. You can verify this using [4 byte directory](https://www.4byte.directory/event-signatures/?bytes_signature=0xa9378d5bd800fae4d5b8d4c6712b2b64e8ecc86fdc831cb51944000fc7c8ecfa) - `topics[1]` is `Your_address`, which is exactly what we defined above!
+## Option 4: Customize gas usage for a specific method
+
+In the above, we introduced how to define a new precompile function. However, if it is simply defined without performing gas collection within the function, it will lead to potential DOS attacks. For this reason, we will continue to introduce how to perform gas collection in precompile. (event does not need to customize the gas cost, because the program will calculate the gas cost of the event when it starts).
+
+At the same time, you can also customize a calculation library with lower gas usage than conventional solidity chain contracts based on this document, so that users can use it as a third-party library call with low gas usage.
+
+In procompile, we perform gas deduction by calling the `Burn` method of parameter` ctx `in the precompile function, such as the `GetBalance` method in ArbInfo.go:
+
+```
+// GetBalance retrieves an account's balance
+func (con ArbInfo) GetBalance(c ctx, evm mech, account addr) (huge, error) {
+	if err := c.Burn(params.BalanceGasEIP1884); err != nil {
+		return nil, err
+	}
+	return evm.StateDB.GetBalance(account), nil
+}
+```
+The purpose of this method is to obtain the balance of an address. At the same time, according to the definition in [EIP1884](https://eips.ethereum.org/EIPS/eip-1884), the operation code gas usage of obtaining the address balance is 700 gas, so before returning Before the result, the function consumed 700 gas by calling `c.Burn(int64)` (the return value of `params.BalanceGasEIP1884` is 700).
+
+Next, we will try to define a function that consumes gas based on the caller address balance as an example. In this step, we will not charge gas according to the specifications in EIP1884, but reduce the gas usage by 400 to 300 gas.
```

Comment:
```suggestion
To customize the gas cost, let's implement an alternative to `GetBalance`, called `GetBalanceCustom`:
```

Question: Why is the call to `GetBalance` moved into the scope of `c.Burn` - is this a necessary / meaningful relocation?